File: js/axios/webhooks/fetchWebhook.js
import axios from "axios";
import "dotenv/config";

const accessKey = process.env.ACCESS_TOKEN;
const apiUrl = process.env.production
  ? "https://api.theauthapi.com"
  : process.env.TESTING_URL;

async function fetchWebhook(webhookId) {
  try {
    const { data } = await axios.get(`${apiUrl}/webhooks/${webhookId}`, {
      headers: {
        ContentType: "application/json",
        "x-api-key": accessKey,
      },
    });
    return data;
  } catch (error) {
    if (error.response) {
      console.log(error.response.data);
    } else {
      // handle other errors
    }
  }
}
(async () => {
  const webhooks = await fetchWebhooks(process.env.PROJECT_ID);
  webhooks.map((id) => {
    console.log("'" + id.id + "',");
  });
  //console.log(webhooks);
})();

File: python/webhooks/fetch_webhook.py
import requests
from requests.exceptions import HTTPError

api_url = "https://api.theauthapi.com/webhooks"
access_key = "REPLACE_WITH_YOUR_ACCESS_KEY"

headers = {
    "x-api-key": access_key,
}

def fetch_webhook(webhook_id):
    try:
        response = requests.get(f'{api_url}/{webhook_id}', headers=headers)
        response.raise_for_status()
        return response.json()
    except HTTPError as error: 
        # handle error
        print(error)


webhook_id = 'REPLACE_WITH_YOUR_WEBHOOK_ID'
webhook = fetch_webhook(webhook_id)
print(webhook)
